lib/utils/FileCache.mjs

```diff
@@ -69,7 +70,7 @@ export default class FileCache {
 		let changed = false;
 
 		if (this._useCheckSum) {
-			changed = configChanged || this.#hasFileCheckSumChanged(descriptor.key, descriptor.meta.hash);
+			changed = configChanged || !metaCache?.hash || metaCache?.hash !== descriptor.meta.hash;
```

Comment:
[suggestion] How about polishing the condition like this? The `metaCache?` repetition seems a bit less readable to me:

```suggestion
			changed = configChanged || !metaCache?.hash || metaCache.hash !== descriptor.meta.hash;
```